openfl/utilities/workspace.py

```diff
@@ -75,3 +78,51 @@ def __exit__(self, exc_type, exc_value, traceback):
         if self.experiment_work_dir in sys.path:
             sys.path.remove(self.experiment_work_dir)
         os.remove(self.data_file_path)
+
+
+def dump_requirements_file(
+        path: Union[str, Path] = './requirements.txt',
+        keep_original_prefixes: bool = True,
+        prefixes: Optional[Union[Tuple[str], str]] = None,
+) -> None:
+    """Prepare and save requirements.txt."""
+    from pip._internal.operations import freeze
+    path = Path(path).absolute()
+
+    # Prepare user provided prefixes for merge with original ones
+    if prefixes is None:
+        prefixes = set()
+    elif type(prefixes) is str:
+        prefixes = set(prefixes,)
+    else:
+        prefixes = set(prefixes)
+
+    # Merge prefixes:
+    # We expect that all the prefixes in a requirement file
+    # are placed at the top
+    if keep_original_prefixes and path.is_file():
+        with open(path) as f:
+            for line in f:
+                if line == '':
+                    continue
```

Comment:
Looks like it is not needed. Or maybe you found corner case for it?

Reply:
This is a mistake. I fixed it to state
```
if line == '\n':
    continue
```
The idea is to skip empty lines between options in the original requirements.txt, for instance:
```
-o option1

-s option2
requirement1==0.0.1
```